### DISCUSSION TITLE
MySQL BINARY(16) column fetch operation throws invalid utf-8 sequence error

### DISCUSSION THREAD

**Discussion by GtenIOS** [rbatis/rbatis#266] [General]
Hello,
I am using `BINARY (16)` column type to store Uuid in my mysql database, and when I try to fetch data to type `rbatis::Bytes`, the library would throw an error `invalid utf-8 sequence of 1 bytes from index 1`

Here's my table structure:
```
CREATE TABLE `test` (
  `id` binary(16) NOT NULL,
  `created` datetime NOT NULL DEFAULT current_timestamp(),
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```

And here's the front end struct from my API server:
```
#[crud_table(table_name:"test")]
pub struct Test {
    pub id: rbatis::Bytes,
}
```

Inserting data works fine: 
```
let uuid: uuid::Uuid = uuid::Uuid::new_v4();
let test: Test = Test {
    id: rbatis::Bytes::from(uuid.as_bytes().to_vec()),
};
rb.save(&test, &[]).await;
```

And here's how I fetch the data:
```
let tests: Vec<Test> = rb.fetch_list().await.unwrap_or_else(|_| vec![]);
```

Upon inspecting the `rbatis-core` lib code, I found that the mysql `BINARY` (& `VARBINARY`) column type is being decoded as `String` instead of `rbatis::Bytes`. Is there a reason why that's the case?

**Comment by zhuxiujia** [comment]
> 

You should use ```rbatis ::Uuid```

**Comment by GtenIOS** [comment]
Doesn't work, in fact it was my first consideration before I went with type `rbatis::Bytes`

Gives me below fetch & insert errors upon using type `rbatis::Uuid` with MySQL column type `Binary(16)`

`ReturnErr  <== invalid utf-8 sequence of 1 bytes from index 1`

`ReturnErr  <== error returned from database: 1406 (22001): Data too long for column 'id' at row 1`

**Comment by GtenIOS** [comment]
The only fix I could find was to change the decoding type of MySQL column type `BINARY(/VARBINARY)` to be `rbatis::Bytes` instead of type `String` from the rbatis core lib
